### PR TITLE
feat(STONEINTG-172): move adapter unit tests to use mock loaders

### PR DESCRIPTION
**Pull request by hongweiliu17**
* Move adapter unit tests to use mock loaders
* Move some function to loader package to make the import smooth

### CODE REVIEW

File: loader/integration.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// GetLatestPipelineRunForSnapshotAndScenario returns the latest Integration PipelineRun for the
+// associated Snapshot and IntegrationTestScenario. In the case the List operation fails,
+// an error will be returned.
+func GetLatestPipelineRunForSnapshotAndScenario(adapterClient client.Client, ctx context.Context, loader ObjectLoader, snapshot *applicationapiv1alpha1.Snapshot, integrationTestScenario *v1alpha1.IntegrationTestScenario) (*tektonv1beta1.PipelineRun, error) {
+	integrationPipelineRuns, err := loader.GetAllPipelineRunsForSnapshotAndScenario(adapterClient, ctx, snapshot, integrationTestScenario)
+	if err != nil {
+		return nil, err
+	}
+
+	var latestIntegrationPipelineRun = &tektonv1beta1.PipelineRun{}
```
[the remark this answers is not in the document]
Comment (hongweiliu17) [reply loader/integration.go:38]: I will investigate and fix it.

**Review by MartinBasti — APPROVED** [review]
Very nice!

File: controllers/pipeline/pipeline_adapter.go [hunk 2]
```diff
@@ -632,3 +613,23 @@ func (a *Adapter) isLatestSucceededPipelineRun() (bool, error) {
 	}
 	return true, nil
 }
+
+// getSucceededBuildPipelineRunsForComponent returns all  succeeded PipelineRun for the
+// associated component. In the case the List operation fails,
+// an error will be returned.
+func getSucceededBuildPipelineRunsForComponent(adapterClient client.Client, ctx context.Context, loader loader.ObjectLoader, component *applicationapiv1alpha1.Component) (*[]tektonv1beta1.PipelineRun, error) {
```
Comment (dirgim) [inline controllers/pipeline/pipeline_adapter.go:620]: Suggestion: dropping the client, context and loader as parameters and using `a` to reference them
```suggestion
func (a *Adapter) getSucceededBuildPipelineRunsForComponent(component *applicationapiv1alpha1.Component) (*[]tektonv1beta1.PipelineRun, error) {
```

File: loader/loader_mock.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+	IntegrationTestScenarioContextKey          contextKey = iota
+	TaskRunContextKey                          contextKey = iota
+	ApplicationComponentsContextKey            contextKey = iota
+	EnvironmentContextKey                      contextKey = iota
+	ReleaseContextKey                          contextKey = iota
+	PipelineRunsContextKey                     contextKey = iota
+	DeploymentTargetClassContextKey            contextKey = iota
+	AllIntegrationTestScenariosContextKey      contextKey = iota
+	RequiredIntegrationTestScenariosContextKey contextKey = iota
+	SnapshotEnvironmentBinding                 contextKey = iota
```
Comment (dirgim) [inline loader/loader_mock.go:57]:
```suggestion
	SnapshotEnvironmentBindingContextKey                 contextKey = iota
```

File: controllers/pipeline/pipeline_adapter.go (hunk 2 above)
Comment (hongweiliu17) [reply controllers/pipeline/pipeline_adapter.go:620]: Agree, thanks!

**Review by jinqi7 — APPROVED** [review]
LGTM

**Review by Josh-Everett — APPROVED** [review]
nice patch!

**Review by dheerajodha — APPROVED** [review]
Thanks for this!

**Comment by hongweiliu17** [comment]
Thanks all for review! Should we merge it after the code freeze? @dirgim

**Pull request merged**